geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -123,9 +127,26 @@ trait KafkaConsumerFeatureCache extends QuadTreeFeatureStore {
     if (geometries.isEmpty) {
       unoptimized(a)
     } else {
-      val envelope = geometries.head.getEnvelopeInternal
-      geometries.tail.foreach(g => envelope.expandToInclude(g.getEnvelopeInternal))
-      new DFR(sft, new DFI(spatialIndex.query(envelope, a.evaluate)))
+      val set = geometries.toSet
```

Comment:
This change might should be considered separately.  Even though it addresses a bug/performance enhancement for the regular KDS in-memory solution.